cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,502 @@
+#!/usr/bin/env python3
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+COP_DOC = """cylc lint [OPTIONS] ARGS
+
+Cylc lint looks through one or more folders for "*.cylc" and "*.rc" files and
+searches for Cylc 7 syntax which may be problematic at Cylc 8.
```

Comment:
@datamel  - what do you think of this:

```suggestion
Cylc lint looks through one or more folders for ".cylc" and ".rc" files.

By default it will examine files for code style. If a workflow has been upgraded
to Cylc 8 by changing the top level `suite.rc` to `flow.cylc` then it will also
check for deprecated Cylc 7 syntax.

You can `cylc lint` with each set of rules alone using `cylc lint -r style`
(for the style checks) or  `cylc lint -r 728` (for the Cylc 7 syntax checks).
```